lib/grpc/client/adapter.ex

```diff
@@ -0,0 +1,23 @@
+defmodule GRPC.Client.Adapter do
+  @moduledoc """
+  HTTP client adapter for GRPC.
+  """
+
+  alias GRPC.Client.Stream
+  alias GRPC.Channel
+
+  @typedoc "Determines if the headers are finished reading."
+  @type fin :: :fin | :nofin
+
+  @callback connect(Channel.t(), map()) :: {:ok, Channel.t()} | {:error, any}
+
+  @callback disconnect(Channel.t()) :: {:ok, Channel.t()} | {:error, any}
+
+  @callback send_request(Stream.t(), binary(), map()) :: Stream.t()
+
+  @callback recv_headers(map(), map(), map()) ::
+              {:ok, %{String.t() => String.t()}, fin} | {:error, GRPC.RPCError.t()}
```

Comment:
Is it better if we just declare fin type here instead of declare at the module level?

Reply:
For `fin` I think we should declare a `@type` because we can have a typedoc specifying what it is and what the values represent